alfasim_sdk/context.py

```diff
@@ -0,0 +1,271 @@
+from enum import Enum
+from typing import List
+from typing import Optional
+
+import attr
+from attr.validators import deep_iterable
+from attr.validators import instance_of
+from attr.validators import optional
+from barril.units import Scalar
+
+from alfasim_sdk._validators import list_of_strings
+from alfasim_sdk._validators import non_empty_str
+
+
+@attr.s(frozen=True)
+class PluginInfo:
+    """
+    PluginInfo provides information about the plugin name, its current state (either enabled or not) and all
+    models defined from this plugin.
+    """
+
+    name = attr.attrib(validator=non_empty_str)
+    enabled = attr.attrib(validator=instance_of(bool))
+    models = attr.attrib(validator=list_of_strings)
+
+
+@attr.s(frozen=True)
+class PipelineSegmentInfo:
+    """
+    The PipelineSegmentInfo provides information about a segments associated with a pipeline.
+
+    PipelineSegmentInfo provides the following attributes:
+        edge_name: name of the edge that the segment is associate with
+
+        start_position: Defines point where this segment starts in MD (measured depth).
+
+        inner_diameter: Inner diameter of pipe annulus.
+
+        roughness: Absolute roughness of the wall of this segment.
+            When ```is_custom``` is true, the reported roughness is customized for this segment,
+            otherwise the reported roughness is the original reported by the wall.
+
+        is_custom: Informs either the roughness value is custom or original from the wall
+    """
+
+    inner_diameter = attr.attrib(validator=instance_of(Scalar))
+    start_position = attr.attrib(validator=instance_of(Scalar))
+    is_custom = attr.attrib(validator=instance_of(bool))
+    roughness = attr.attrib(validator=instance_of(Scalar))
+
+
+list_of_segments = deep_iterable(
+    member_validator=instance_of(PipelineSegmentInfo),
+    iterable_validator=instance_of(list),
+)
+
+
+@attr.s(frozen=True)
+class PipelineInfo:
+    """
+    The PipelineInfo provides information about the geometry of a pipeline.
+
+    PipelineSegmentInfo provides the following attributes:
+        name: Name associate with this ``Pipeline`` on ALFAsim
+
+        edge_name: Name of the edge that this ``Pipeline`` is associate with.
+
+        segments: List of segments associates with this ``Pipeline``
+        For more information check `alfasim_sdk.context.PipelineSegmentInfo`
+
+        total_length: Total length of the pipeline.
+
+    """
+
+    name = attr.attrib(type=str, validator=non_empty_str)
```

Comment:
I'm not sure Pipeline names in ALFAsim will ever be different than "<None>".

Also:
Name associated* with...
``Pipeline`` is associated* with.

Reply:
Commit 87ba400 fix the grammar :)

Reply:
And what about the name property?

Reply:
Actually I just checked and they are named after the edge's name: "Edge 1 > Pipeline" or "My Custom Edge Name > Pipeline". But this is also an implementation detail, you can leave the _name_ in this class.